test/functional/rpc_decodescript.py

```diff
@@ -96,6 +96,16 @@ def decodescript_script_pub_key(self):
         assert_equal('witness_v0_scripthash', rpc_result['segwit']['type'])
         assert_equal('0 ' + multisig_script_hash, rpc_result['segwit']['asm'])
 
+        # Check decoding a multisig redeem script doesn't return an invalid descriptor
+        self.log.info("- multisig invalid")
+        # Decode a +520 bytes consensus-invalid p2sh multisig - provided script translates to multi(20, keys)
+        large_multisig_script = '0114' + push_public_key * 20 + '0114' + 'ae'
+        rpc_result = self.nodes[0].decodescript(large_multisig_script)
+        # Verify it is valid only under segwit rules, not for bare multisig nor p2sh.
+        assert 'desc' not in rpc_result
```

Comment:
> ```
> assert_equal('multisig', rpc_result['type'])
> ```
> Nit: This assertion can also be added. I modified the test case to have 21 keys, and the `type` changed to `nonstandard`, which is correct. `p2sh` and `desc` values are present in that output.
> 
> Do you think it's worth adding the test case with 21 pubkeys as well to cover the `nonstandard` scenario?

Hmm, I find the `type` return field misleading. Based on the code, it relates to the output script type, which in this case can only be represented as p2wsh and not as a bare multisig. This is because (1) bare multisig scripts with +3 pubkeys are non-standard, and (2) bare multisig scripts with +15 pubkeys are consensus-invalid.

So, I'm not entirely convinced about the type assertion at this point. It seems we should improve the type field RPC description, correct the RPC to actually return "non-standard" for bare multisigs with +3 pubkeys, and perhaps introduce the "consensus-invalid" (or simply "invalid") `type` too.